librz/bin/bobj.c

```diff
@@ -315,21 +315,25 @@ RZ_API int rz_bin_object_set_items(RzBinFile *bf, RzBinObject *o) {
 	if (p->fields) {
 		o->fields = p->fields(bf);
 		if (o->fields) {
-			o->fields->free = rz_bin_field_free;
+			if (!o->fields->free) {
```

Comment:
Where does this happen? I'd say that plugins are expected to pass a RzList that is properly created with the right free callback. If this does not happen, it's the plugin's fault, not of RzBin. WDYT?

Reply:
i prefer to have this fail fallback or do you want to have `rz_warn_if_fail()`?

Reply:
I prefer `rz_warn_if_fail`. Plugins should really be fixed, we should not silently "fixing them".